crates/consensus/src/request.rs

```diff
@@ -4,7 +4,42 @@ use alloy_eips::{
     eip7251::ConsolidationRequest,
     eip7685::{Decodable7685, Eip7685Error, Encodable7685},
 };
-use alloy_rlp::{Decodable, Encodable};
+use alloy_primitives::Bytes;
+use alloy_rlp::{BufMut, Decodable, Encodable};
+
+/// A list of EIP-7685 requests.
+#[derive(Debug, Clone, PartialEq, Eq, Default, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
+pub struct Requests(pub Vec<Request>);
+
+impl Encodable for Requests {
+    fn encode(&self, out: &mut dyn BufMut) {
+        let mut h = alloy_rlp::Header { list: true, payload_length: 0 };
+
+        let mut encoded = Vec::new();
+        for req in &self.0 {
+            let encoded_req = req.encoded_7685();
+            h.payload_length += encoded_req.len();
+            encoded.push(Bytes::from(encoded_req));
+        }
+
+        h.encode(out);
+        for req in encoded {
+            req.encode(out);
+        }
+    }
+}
+
+impl Decodable for Requests {
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        Ok(<Vec<Bytes> as Decodable>::decode(buf)?
+            .into_iter()
+            .map(|bytes| Request::decode_7685(&mut bytes.as_ref()))
+            .collect::<Result<Vec<_>, alloy_eips::eip7685::Eip7685Error>>()
+            .map(Self)?)
+    }
+}
```

Comment:
pls add tests for these impls!

Reply:
There already was a Requests object upstream :)

Reply:
I've removed this now